hypothesis_bio/hypothesis_bio.py

```diff
@@ -241,43 +242,67 @@ def fasta(
     - `sequence_source`: The source of the sequence. Defaults to [`dna`](#dna).
     - `wrap_length`: The width to wrap the sequence on. If `None`, mixed sizes are used.
     - `allow_windows_line_endings`: Whether to allow `\\r\\n` in the linebreaks.
+    = `multiple_fasta`: Whether multiple FASTA sequences will be generated.
     """
     if comment_source is None:
         comment_source = text(alphabet=characters(min_codepoint=32, max_codepoint=126))
     if sequence_source is None:
         sequence_source = dna()
 
-    comment = draw(comment_source)
-    sequence = draw(sequence_source)
+    if num_fasta is None:
+        # randomly determine how many FASTA sequences to generate
+        num_fasta = draw(integers(min_value=0)) ##replace with numpy if needed
+
+    # Lists to hold comments and sequences for each FASTA sequence
+    comments = []
+    sequences = []
+    # generate pre-determined number of sequences
+    for i in range(num_fasta):
+        comments.append(draw(comment_source))
+        sequences.append(draw(sequence_source))
 
     # the nice case where the user gave the wrap size
     if wrap_length is not None:
-        sequence = fill(sequence, wrap_length, break_on_hyphens=False)
+        for sequence in sequences:
+            sequence = fill(sequence, wrap_length, break_on_hyphens=False)
 
     # the pathological case
     elif wrap_length is None:
 
-        # choose where to wrap
-        indices = [
-            draw(integers(min_value=0, max_value=len(sequence)))
-            for i in range(draw(integers(min_value=0, max_value=len(sequence))))
-        ]
-        indices = list(set(indices))
+        # list stores lists of indices for each sequences
+        allseq_indices = []
+
+        for sequence in sequences:
+            # choose where to wrap
+            indices = [
+                draw(integers(min_value=0, max_value=len(sequence)))
+                for i in range(draw(integers(min_value=0, max_value=len(sequence))))
+                ]
+            indices = list(set(indices))
```

Comment:
Switch the list comprehension to [set comprehension](https://python-reference.readthedocs.io/en/latest/docs/comprehensions/set_comprehension.html) then you don't have to turn a list into a set to turn it back into a list again.